ROMUtils.h

```diff
@@ -19,6 +19,7 @@ namespace ROMUtils
     extern unsigned int CurrentFileSize;
     extern QString ROMFilePath;
     extern unsigned int SaveDataIndex;
+    extern LevelComponents::Tileset *singletonTilesets[92];
```

Comment:
WHY IS THIS A POINTER??????

Reply:
+1;  let's try to do a smart pointer. If it's a pointer because it's the underlying type for some instance variable somewhere, let's change that as well to remove the pointers. If we use fewer pointers, we'll probably cut down on memory leaks as well.

Reply:
@Goldensunboy It's supposed to be a std::vector type instead. Most of the pointers I see in the code can be replaced with std::vector, smart pointers, references, or objects.